repos/spack_repo/builtin/packages/py_mumps4py/package.py

```diff
@@ -0,0 +1,63 @@
+from spack_repo.builtin.build_systems.python import PythonPackage
+
+from spack.package import *
+
+
+class PyMumps4py(PythonPackage):
+    """Python wrapper for the MUMPS solver (MUMPS4PY)."""
+
+    homepage = "https://github.com/imadki/mumps4py"
+    url = "https://github.com/imadki/mumps4py/archive/refs/tags/1.0.0.tar.gz"
+
+    maintainers = ["williampiat3"]
+
+    version("1.0.0", sha256="634dd52a9942e88a430d6c8b519cbf6e50db77dd5f1be1cb04e5ab3f3e9da8ba")
+
+    # Python requirements
+    depends_on("python@3.5:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    depends_on("py-pip", type="build")
+
+    # Runtime and build deps
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-cython", type="build")
+    depends_on("py-mpi4py", type=("build", "run"))
+
+    # Optional/test deps
+    depends_on("py-pytest", type="test")
+
+    # External solver
+    depends_on("mumps")
+    depends_on("cmake", type="build")
+
+    def build_args(self, spec, prefix):
+        # Ensure MUMPS include/lib are passed if setup.py needs them
+        mumps = spec["mumps"]
+        args = [
+            "--inplace",
+            "MUMPS_INC={0}".format(mumps.prefix.include),
+            "MUMPS_LIB={0}".format(mumps.prefix.lib),
+        ]
+        return args
+
+    def setup_build_environment(self, env):
+        # Required by mumps4py to specify which MUMPS solvers to use
+        env.set("MUMPS_SOLVERS", "dmumps,cmumps,zmumps,smumps")
```

Comment:
Does this list not depend at all on how mumps itself was built?

Reply:
By default, mumps builds all 4 solvers, this env variable allows to build the python bindings for each mumps solver. This usually allow users to adapt the build for custom install, but through spack we want to install all solver bindings.

Reply:
I think there needs to be a query here to check how `mumps` was built, and set the solvers based on how mumps was built.

I'm not a mumps expert, but something like this is what I would expect:

```suggestion
    def setup_build_environment(self, env):
        solvers = []
        mumps = self.spec["mumps"]
        if mumps.satisfies("+float"):
            solvers.append("smumps")
        if mumps.satisfies("+double"):
            solvers.append("dmumps")
        if mumps.satisfies("+float+complex"):
            solvers.append("cmumps")
        if mumps.satisfies("+double+complex"):
            solvers.append("zmumps")

        # Required by mumps4py to specify which MUMPS solvers to use
        env.set("MUMPS_SOLVERS", ",".join(solvers))
```
@jcortial-safran you're listed as the maintainer for `mumps`, do you think that's the right way to figure out which solvers are available?

Reply:
IMHO we should not query mumps to know what solvers are available. Rather, mumps4py should mandate which solvers must be available in MUMPS.

By default, all 4 solvers are built in the [MUMPS spack recipe](https://github.com/spack/spack-packages/blob/develop/repos/spack_repo/builtin/packages/mumps/package.py) so `requires("mumps")` is already enough to ensure that the prerequisite for `env.set("MUMPS_SOLVERS", "dmumps,cmumps,zmumps,smumps")` is fulfilled. And if you really want to be sure, you can explicitly specify `requires("mumps+float+complex+double")`.

If you *really* want to allow mumps4py to work with a subset of the MUMPS solvers (an approach that I would not recommend btw), you could  for instance mirror the `float`, `double` and `complex` variants of MUMPS in the mumps4py recipe, and add the corresponding prerequisites, e.g. `requires("mumps+double", when="+double")` and so on.

Reply:
@becker33, do you think I should complexify the recipe to allow working with subsets of MUMPS solver?

Reply:
For now I think it's fine to just say `depends_on("mumps+float+complex+double")` and keep the solver list simple.